apps/mobile/src/components/wallet-manager/account-list.tsx

```diff
@@ -0,0 +1,32 @@
+import { ReactNode } from 'react';
+import { Pressable, Text, View } from 'react-native';
+
+import { Account } from '@/state/accounts/accounts';
+
+interface AccountsLayoutProps {
+  fingerprint: string;
+  accounts: Account[];
+  onRemoveAccount: (accountIndex: number) => void;
+  renderKeychains(fingerprint: string, accountIndex: number): ReactNode;
+}
+export function AccountsLayout({
+  fingerprint,
+  accounts,
+  onRemoveAccount,
+  renderKeychains,
+}: AccountsLayoutProps) {
+  return accounts.map(account => (
+    <View key={account.id} style={{ marginBottom: 20 }}>
+      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+        <Text style={{ fontSize: 16, marginBottom: 8 }}>Account {account.accountIndex}</Text>
+        <Pressable
+          style={{ marginBottom: 12, marginLeft: 8 }}
+          onPress={() => onRemoveAccount(account.accountIndex)}
+        >
+          <Text>❌</Text>
+        </Pressable>
+      </View>
+      {renderKeychains(fingerprint, account.accountIndex)}
```

Comment:
Some blog posts will say: _😡 why u use render props, hooks replace need entirely in every possible use case._ IMO this is still a clean pattern and creates a nice separation between high level state-consuming components from dumb components.

Reply:
I agree and although hooks have their place they can be over-used